jsonrpc/src/main/scala/org/langmeta/jsonrpc/MessageWriter.scala

```diff
@@ -34,13 +35,16 @@ class MessageWriter(out: OutputStream) extends LazyLogging {
     lock.synchronized {
       require(h.get(ContentLen).isEmpty)
 
-      val str = msg.asJson.noSpaces
+      val json = msg.asJson.withObject(
+        obj => Json.fromJsonObject(obj.add("jsonrpc", Json.fromString("2.0")))
+      )
```

Comment:
Nice! Didn't know about `"2.0".asJson`. I went with 

```
val json = msg.asJson.withObject(_.add("jsonrpc", "2.0".asJson).asJson)
```

since we only want to add this if it's an object.

I find it a bit annoying to require `.asJson` on JsonObject, do you have an intuition why it's like this?

Reply:
I guess JsonObject is a complex data structure that it's unlreated to the Json AST per se. Anyway, I'm from the phone so I can't try, but I think mapObject would do the trick, since it's from JSONObject => JSONObject